src/uwtools/loaders.py

```diff
@@ -3,24 +3,20 @@
 
 https://pyyaml.org/wiki/PyYAMLDocumentation
 """
-
+import os
 import collections
 import pathlib
 from typing import Any, Dict
-from yaml import load
-
-try:
-    from yaml import CLoader as Loader
-except ImportError:
-    from yaml import Loader
 
+from uwtools.yaml_file import YAMLFile
 
 class Config(collections.UserDict):
     def __getattr__(self, name):
         return self.__dict__["data"][name]
 
-
-def load_yaml(_path: pathlib.Path):
-    with open(_path, "r") as _file:
-        props = load(_file, Loader=Loader)
-        return Config(props)
+def load_yaml(config_file: pathlib.Path):
+    if config_file is not None:
+        config = YAMLFile(os.path.abspath(config_file))
+        if config is None:
+            config = {}
```

Comment:
These two paths will return different types of data structures. Do we want that?

Reply:
Empty Dict or Dict and in the Class using NiceDict is this what you mean two paths?

Reply:
I mean, you will get a YAMLFile object when you pass a config, but a python dict object when you pass a config file.

Reply:
I thought YAMLFILE objects returned NiceDict objects but I can be wrong, let me double check.

Reply:
Oh wait I recall it returns dict and the class it is in overloads it to NiceDict

Reply:
Let me do an inspection on the returning object to double check

Reply:
Update function call (not the Class Method anymore) to always return NiceDict object with rational documented in the code remarks